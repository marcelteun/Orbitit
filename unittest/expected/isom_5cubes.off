OFF
#
# isom_5cubes
#
# file generated with python script by Marcel Tunnissen
# Vertices Faces Edges
40 30 60
# Vertices
-0.000000000000118 -0.618033988749882 1.618033988750117
-1.618033988750117 0.000000000000118 0.618033988749882
-1.000000000000117 -1.000000000000117 -1.000000000000117
0.618033988749882 -1.618033988750117 0.000000000000118
1.000000000000117 1.000000000000117 1.000000000000117
-0.618033988749882 1.618033988750117 -0.000000000000118
0.000000000000118 0.618033988749882 -1.618033988750117
1.618033988750117 -0.000000000000118 -0.618033988749882
1.618033988750117 0.000000000000118 0.618033988749882
0.618033988749882 1.618033988750117 -0.000000000000118
-1.000000000000117 1.000000000000117 1.000000000000117
0.000000000000118 -0.618033988749882 1.618033988750117
1.000000000000117 -1.000000000000117 -1.000000000000117
-0.000000000000118 0.618033988749882 -1.618033988750117
-1.618033988750117 -0.000000000000118 -0.618033988749882
-0.618033988749882 -1.618033988750117 0.000000000000118
-1 -1 1
-1 1 1
-1 1 -1
-1 -1 -1
1 -1 1
1 1 1
1 1 -1
1 -1 -1
-1.618033988750118 -0.000000000000117 0.618033988749882
-0.618033988749883 -1.618033988750117 -0.000000000000118
1.000000000000117 -1.000000000000118 1.000000000000117
-0.000000000000118 0.618033988749882 1.618033988750117
-1.000000000000117 1.000000000000118 -1.000000000000117
0.000000000000118 -0.618033988749882 -1.618033988750117
1.618033988750118 0.000000000000117 -0.618033988749882
0.618033988749883 1.618033988750117 0.000000000000118
-1.618033988750117 0.000000000000117 -0.618033988749882
-0.618033988749882 1.618033988750117 0.000000000000118
1.000000000000117 1.000000000000118 -1.000000000000117
-0.000000000000118 -0.618033988749882 -1.618033988750117
-1.000000000000117 -1.000000000000118 1.000000000000117
0.000000000000118 0.618033988749882 1.618033988750117
1.618033988750117 -0.000000000000117 0.618033988749882
0.618033988749882 -1.618033988750117 -0.000000000000118
# Sides and colours
4  0 1 2 3  0.996094 0.839844 0.0
4  0 3 7 4  0.996094 0.839844 0.0
4  1 0 4 5  0.996094 0.839844 0.0
4  2 1 5 6  0.996094 0.839844 0.0
4  3 2 6 7  0.996094 0.839844 0.0
4  7 6 5 4  0.996094 0.839844 0.0
4  8 9 10 11  0.132812 0.542969 0.132812
4  8 11 15 12  0.132812 0.542969 0.132812
4  9 8 12 13  0.132812 0.542969 0.132812
4  10 9 13 14  0.132812 0.542969 0.132812
4  11 10 14 15  0.132812 0.542969 0.132812
4  15 14 13 12  0.132812 0.542969 0.132812
4  16 17 18 19  0.542969 0.0 0.0
4  16 19 23 20  0.542969 0.0 0.0
4  17 16 20 21  0.542969 0.0 0.0
4  18 17 21 22  0.542969 0.0 0.0
4  19 18 22 23  0.542969 0.0 0.0
4  23 22 21 20  0.542969 0.0 0.0
4  24 25 26 27  0.0 0.746094 0.996094
4  24 27 31 28  0.0 0.746094 0.996094
4  25 24 28 29  0.0 0.746094 0.996094
4  26 25 29 30  0.0 0.746094 0.996094
4  27 26 30 31  0.0 0.746094 0.996094
4  31 30 29 28  0.0 0.746094 0.996094
4  32 33 34 35  0.542969 0.523438 0.304688
4  32 35 39 36  0.542969 0.523438 0.304688
4  33 32 36 37  0.542969 0.523438 0.304688
4  34 33 37 38  0.542969 0.523438 0.304688
4  35 34 38 39  0.542969 0.523438 0.304688
4  39 38 37 36  0.542969 0.523438 0.304688
# END
