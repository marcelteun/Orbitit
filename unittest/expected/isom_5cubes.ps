%!PS-Adobe-2.0
%%Title: isom_5cubes
%%Creator: PS.py written by marcelteun
%%Pages: 1 1
%%BoundingBox: 0 0 595 842
%%EndComments
%%BeginSetup
%%EndSetup
%%Page: 1 1
%%BeginPage
10.000000 10.000000 translate
/vertices [
  [1 1] % 0
  [-1 1] % 1
  [-1 -1] % 2
  [1 -1] % 3
  [0.236068 -1] % 4
  [1 1] % 5
  [1 1] % 6
  [-1 -0.236068] % 7
  [-1 -0.236068] % 8
  [0.236068 -1] % 9
  [-0.236068 1] % 10
  [-1 -1] % 11
  [-1 -1] % 12
  [1 0.236068] % 13
  [1 0.236068] % 14
  [-0.236068 1] % 15
  [1 -0.236068] % 16
  [-1 1] % 17
  [-1 1] % 18
  [-0.236068 -1] % 19
  [-0.236068 -1] % 20
  [1 -0.236068] % 21
  [-1 0.236068] % 22
  [1 -1] % 23
  [0.236068 1] % 24
  [-1 0.236068] % 25
  [1 -1] % 26
  [0.236068 1] % 27
] def
/faces [
  [0 1 2 3][4 5][6 7][8 9][10 11][12 13][14 15][16 17][18 19][20 21][22 23]
  [24 25][26 27]
] def
/bbox [-1 -1 1 1] def
/scalingSize 50 def
.1 setlinewidth

/dx bbox 2 get bbox 0 get sub scalingSize mul def
/dy bbox 3 get bbox 1 get sub scalingSize mul def

gsave
  bbox 0 get scalingSize mul neg
  bbox 1 get scalingSize mul neg
  translate
  faces aload length { %repeat
    dup 0 get
    vertices exch get
    aload pop
    scalingSize mul exch
    scalingSize mul exch
    moveto
    aload length 1 sub { %repeat
      vertices exch get
      aload pop
      scalingSize mul exch
      scalingSize mul exch
      lineto
    } repeat
    pop
    closepath stroke
  } repeat


grestore
105.000000 0 translate
/vertices [
  [-1.4120227 -0.0786893] % 0
  [0.0786893 -1.4120227] % 1
  [-0.0786893 1.4120227] % 2
  [1.4120227 0.0786893] % 3
  [0.490712 0.9027346] % 4
  [-1.4120227 -0.0786893] % 5
  [-1.4120227 -0.0786893] % 6
  [0.9027346 -0.490712] % 7
  [0.9027346 -0.490712] % 8
  [0.490712 0.9027346] % 9
  [-0.8426213 -0.5879773] % 10
  [0.5879773 -0.8426213] % 11
  [-0.0786893 1.4120227] % 12
  [-0.8426213 -0.5879773] % 13
  [0.5879773 -0.8426213] % 14
  [-0.0786893 1.4120227] % 15
  [-0.9027346 0.490712] % 16
  [-0.490712 -0.9027346] % 17
  [-0.490712 -0.9027346] % 18
  [1.4120227 0.0786893] % 19
  [1.4120227 0.0786893] % 20
  [-0.9027346 0.490712] % 21
  [0.8426213 0.5879773] % 22
  [-0.5879773 0.8426213] % 23
  [0.0786893 -1.4120227] % 24
  [0.8426213 0.5879773] % 25
  [-0.5879773 0.8426213] % 26
  [0.0786893 -1.4120227] % 27
] def
/faces [
  [0 1 3 2][4 5][6 7][8 9][10 11][12 13][14 15][16 17][18 19][20 21][22 23]
  [24 25][26 27]
] def
/bbox [-1.4120227 -1.4120227 1.4120227 1.4120227] def
/scalingSize 50 def
.1 setlinewidth

/dx bbox 2 get bbox 0 get sub scalingSize mul def
/dy bbox 3 get bbox 1 get sub scalingSize mul def

gsave
  bbox 0 get scalingSize mul neg
  bbox 1 get scalingSize mul neg
  translate
  faces aload length { %repeat
    dup 0 get
    vertices exch get
    aload pop
    scalingSize mul exch
    scalingSize mul exch
    moveto
    aload length 1 sub { %repeat
      vertices exch get
      aload pop
      scalingSize mul exch
      scalingSize mul exch
      lineto
    } repeat
    pop
    closepath stroke
  } repeat


grestore
146.202266 0 translate
/vertices [
  [0.4472136 -1.3416408] % 0
  [-1.3416408 -0.4472136] % 1
  [1.3416408 0.4472136] % 2
  [-0.4472136 1.3416408] % 3
  [0.4472136 -1.3416408] % 4
  [0.236068 1] % 5
  [0.236068 1] % 6
  [-1 0.236068] % 7
  [-1 0.236068] % 8
  [0.4472136 -1.3416408] % 9
  [-0.7888544 0.6583592] % 10
  [-0.6583592 -0.7888544] % 11
  [-0.6583592 -0.7888544] % 12
  [1.3416408 0.4472136] % 13
  [1.3416408 0.4472136] % 14
  [-0.7888544 0.6583592] % 15
  [-1.3416408 -0.4472136] % 16
  [0.7888544 -0.6583592] % 17
  [0.6583592 0.7888544] % 18
  [-1.3416408 -0.4472136] % 19
  [0.7888544 -0.6583592] % 20
  [0.6583592 0.7888544] % 21
  [-0.4472136 1.3416408] % 22
  [-0.236068 -1] % 23
  [1 -0.236068] % 24
  [-0.4472136 1.3416408] % 25
  [-0.236068 -1] % 26
  [1 -0.236068] % 27
] def
/faces [
  [1 0 2 3][4 5][6 7][8 9][10 11][12 13][14 15][16 17][18 19][20 21][22 23]
  [24 25][26 27]
] def
/bbox [-1.3416408 -1.3416408 1.3416408 1.3416408] def
/scalingSize 50 def
.1 setlinewidth

/dx bbox 2 get bbox 0 get sub scalingSize mul def
/dy bbox 3 get bbox 1 get sub scalingSize mul def

gsave
  bbox 0 get scalingSize mul neg
  bbox 1 get scalingSize mul neg
  translate
  faces aload length { %repeat
    dup 0 get
    vertices exch get
    aload pop
    scalingSize mul exch
    scalingSize mul exch
    moveto
    aload length 1 sub { %repeat
      vertices exch get
      aload pop
      scalingSize mul exch
      scalingSize mul exch
      lineto
    } repeat
    pop
    closepath stroke
  } repeat


grestore
139.164079 0 translate
/vertices [
  [1 -1] % 0
  [-1 -1] % 1
  [1 1] % 2
  [-1 1] % 3
  [1 -0.236068] % 4
  [-1 1] % 5
  [-0.236068 -1] % 6
  [1 -0.236068] % 7
  [-1 1] % 8
  [-0.236068 -1] % 9
  [-1 -1] % 10
  [1 0.236068] % 11
  [1 0.236068] % 12
  [-0.236068 1] % 13
  [-0.236068 1] % 14
  [-1 -1] % 15
  [1 -1] % 16
  [0.236068 1] % 17
  [-1 0.236068] % 18
  [1 -1] % 19
  [0.236068 1] % 20
  [-1 0.236068] % 21
  [1 1] % 22
  [-1 -0.236068] % 23
  [-1 -0.236068] % 24
  [0.236068 -1] % 25
  [0.236068 -1] % 26
  [1 1] % 27
] def
/faces [
  [1 0 2 3][4 5][6 7][8 9][10 11][12 13][14 15][16 17][18 19][20 21][22 23]
  [24 25][26 27]
] def
/bbox [-1 -1 1 1] def
/scalingSize 50 def
.1 setlinewidth

/dx bbox 2 get bbox 0 get sub scalingSize mul def
/dy bbox 3 get bbox 1 get sub scalingSize mul def

gsave
  bbox 0 get scalingSize mul neg
  bbox 1 get scalingSize mul neg
  translate
  faces aload length { %repeat
    dup 0 get
    vertices exch get
    aload pop
    scalingSize mul exch
    scalingSize mul exch
    moveto
    aload length 1 sub { %repeat
      vertices exch get
      aload pop
      scalingSize mul exch
      scalingSize mul exch
      lineto
    } repeat
    pop
    closepath stroke
  } repeat


grestore
105.000000 0 translate
-495.366345 146.202266 translate
/vertices [
  [-1 -1] % 0
  [-1 1] % 1
  [1 -1] % 2
  [1 1] % 3
  [0.236068 1] % 4
  [-1 0.236068] % 5
  [1 -1] % 6
  [0.236068 1] % 7
  [-1 0.236068] % 8
  [1 -1] % 9
  [-0.236068 1] % 10
  [-1 -1] % 11
  [-1 -1] % 12
  [1 0.236068] % 13
  [1 0.236068] % 14
  [-0.236068 1] % 15
  [-1 -0.236068] % 16
  [0.236068 -1] % 17
  [1 1] % 18
  [-1 -0.236068] % 19
  [0.236068 -1] % 20
  [1 1] % 21
  [-0.236068 -1] % 22
  [1 -0.236068] % 23
  [-1 1] % 24
  [-0.236068 -1] % 25
  [1 -0.236068] % 26
  [-1 1] % 27
] def
/faces [
  [1 0 2 3][4 5][6 7][8 9][10 11][12 13][14 15][16 17][18 19][20 21][22 23]
  [24 25][26 27]
] def
/bbox [-1 -1 1 1] def
/scalingSize 50 def
.1 setlinewidth

/dx bbox 2 get bbox 0 get sub scalingSize mul def
/dy bbox 3 get bbox 1 get sub scalingSize mul def

gsave
  bbox 0 get scalingSize mul neg
  bbox 1 get scalingSize mul neg
  translate
  faces aload length { %repeat
    dup 0 get
    vertices exch get
    aload pop
    scalingSize mul exch
    scalingSize mul exch
    moveto
    aload length 1 sub { %repeat
      vertices exch get
      aload pop
      scalingSize mul exch
      scalingSize mul exch
      lineto
    } repeat
    pop
    closepath stroke
  } repeat


grestore
105.000000 0 translate
/vertices [
  [-0.4472136 -1.3416408] % 0
  [-1.3416408 0.4472136] % 1
  [0.4472136 1.3416408] % 2
  [1.3416408 -0.4472136] % 3
  [-1 -0.236068] % 4
  [0.236068 -1] % 5
  [0.236068 -1] % 6
  [0.4472136 1.3416408] % 7
  [0.4472136 1.3416408] % 8
  [-1 -0.236068] % 9
  [-0.236068 1] % 10
  [-0.4472136 -1.3416408] % 11
  [1 0.236068] % 12
  [-0.236068 1] % 13
  [-0.4472136 -1.3416408] % 14
  [1 0.236068] % 15
  [-0.6583592 0.7888544] % 16
  [-0.7888544 -0.6583592] % 17
  [-0.7888544 -0.6583592] % 18
  [1.3416408 -0.4472136] % 19
  [1.3416408 -0.4472136] % 20
  [-0.6583592 0.7888544] % 21
  [0.7888544 0.6583592] % 22
  [-1.3416408 0.4472136] % 23
  [-1.3416408 0.4472136] % 24
  [0.6583592 -0.7888544] % 25
  [0.6583592 -0.7888544] % 26
  [0.7888544 0.6583592] % 27
] def
/faces [
  [3 2 1 0][4 5][6 7][8 9][10 11][12 13][14 15][16 17][18 19][20 21][22 23]
  [24 25][26 27]
] def
/bbox [-1.3416408 -1.3416408 1.3416408 1.3416408] def
/scalingSize 50 def
.1 setlinewidth

/dx bbox 2 get bbox 0 get sub scalingSize mul def
/dy bbox 3 get bbox 1 get sub scalingSize mul def

gsave
  bbox 0 get scalingSize mul neg
  bbox 1 get scalingSize mul neg
  translate
  faces aload length { %repeat
    dup 0 get
    vertices exch get
    aload pop
    scalingSize mul exch
    scalingSize mul exch
    moveto
    aload length 1 sub { %repeat
      vertices exch get
      aload pop
      scalingSize mul exch
      scalingSize mul exch
      lineto
    } repeat
    pop
    closepath stroke
  } repeat


grestore
139.164079 0 translate

showpage
%%EndPage

%%Tailer
%%EOF
