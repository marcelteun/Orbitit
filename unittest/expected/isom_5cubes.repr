orbitit.geom_3d.SymmetricShape(
    vs=[
        orbitit.geomtypes.Vec3([1, 1, 1]),
        orbitit.geomtypes.Vec3([-1, 1, 1]),
        orbitit.geomtypes.Vec3([-1, -1, 1]),
        orbitit.geomtypes.Vec3([1, -1, 1]),
        orbitit.geomtypes.Vec3([1, 1, -1]),
        orbitit.geomtypes.Vec3([-1, 1, -1]),
        orbitit.geomtypes.Vec3([-1, -1, -1]),
        orbitit.geomtypes.Vec3([1, -1, -1])
    ],
    fs=[
        [0, 1, 2, 3],
        [0, 3, 7, 4],
        [1, 0, 4, 5],
        [2, 1, 5, 6],
        [3, 2, 6, 7],
        [7, 6, 5, 4]
    ],
    es=[0, 1, 1, 2, 2, 3, 0, 3, 3, 7, 4, 7, 0, 4, 4, 5, 1, 5, 5, 6, 2, 6, 6, 7],
    colors=[
        [0.996094, 0.839844, 0.0],
        [0.132812, 0.542969, 0.132812],
        [0.542969, 0.0, 0.0],
        [0.0, 0.746094, 0.996094],
        [0.542969, 0.523438, 0.304688]
    ],
    isometries=[
        orbitit.geomtypes.Rot3((
            orbitit.geomtypes.Quat([-0.809016994375, -0.5, 0.309016994375, 0]),
            orbitit.geomtypes.Quat([-0.809016994375, 0.5, -0.309016994375, 0]),
        )),
        orbitit.geomtypes.Rot3((
            orbitit.geomtypes.Quat([0.809016994375, -0.309016994375, 0, -0.5]),
            orbitit.geomtypes.Quat([0.809016994375, 0.309016994375, 0, 0.5]),
        )),
        orbitit.geomtypes.Rot3((
            orbitit.geomtypes.Quat([0.5, 0.5, -0.5, -0.5]),
            orbitit.geomtypes.Quat([0.5, -0.5, 0.5, 0.5]),
        )),
        orbitit.geomtypes.Rot3((
            orbitit.geomtypes.Quat([0.5, 0, -0.309016994375, 0.809016994375]),
            orbitit.geomtypes.Quat([0.5, 0, 0.309016994375, -0.809016994375]),
        )),
        orbitit.geomtypes.Rot3((
            orbitit.geomtypes.Quat([0, -0.5, 0.809016994375, 0.309016994375]),
            orbitit.geomtypes.Quat([0, 0.5, -0.809016994375, -0.309016994375]),
        ))
    ],
    name='isom_5cubes',
    orientation=orbitit.geomtypes.Rot3((
        orbitit.geomtypes.Quat([1, 0, 0, 0]),
        orbitit.geomtypes.Quat([1, 0, 0, 0]),
    ))
)
