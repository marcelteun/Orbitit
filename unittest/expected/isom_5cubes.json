{"class": "symmetric_shape", "data": {"cols": [[0.996094, 0.839844, 0], [0.132812, 0.542969, 0.132812], [0.542969, 0, 0], [0, 0.746094, 0.996094], [0.542969, 0.523438, 0.304688]], "fs": [[0, 1, 2, 3], [0, 3, 7, 4], [1, 0, 4, 5], [2, 1, 5, 6], [3, 2, 6, 7], [7, 6, 5, 4]], "isometries": [{"class": "transform_3d", "data": [{"class": "quaternion", "data": [-0.809016994375, -0.5, 0.309016994375, 0]}, {"class": "quaternion", "data": [-0.809016994375, 0.5, -0.309016994375, 0]}]}, {"class": "transform_3d", "data": [{"class": "quaternion", "data": [0.809016994375, -0.309016994375, 0, -0.5]}, {"class": "quaternion", "data": [0.809016994375, 0.309016994375, 0, 0.5]}]}, {"class": "transform_3d", "data": [{"class": "quaternion", "data": [0.5, 0.5, -0.5, -0.5]}, {"class": "quaternion", "data": [0.5, -0.5, 0.5, 0.5]}]}, {"class": "transform_3d", "data": [{"class": "quaternion", "data": [0.5, 0, -0.309016994375, 0.809016994375]}, {"class": "quaternion", "data": [0.5, 0, 0.309016994375, -0.809016994375]}]}, {"class": "transform_3d", "data": [{"class": "quaternion", "data": [0, -0.5, 0.809016994375, 0.309016994375]}, {"class": "quaternion", "data": [0, 0.5, -0.809016994375, -0.309016994375]}]}], "name": "isom_5cubes", "orientation": {"class": "transform_3d", "data": [{"class": "quaternion", "data": [1, 0, 0, 0]}, {"class": "quaternion", "data": [1, 0, 0, 0]}]}, "vs": [[1, 1, 1], [-1, 1, 1], [-1, -1, 1], [1, -1, 1], [1, 1, -1], [-1, 1, -1], [-1, -1, -1], [1, -1, -1]]}}